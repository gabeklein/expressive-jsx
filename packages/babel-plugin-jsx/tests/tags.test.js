transform("strip tag; emit default", `
  () => {
    <hello>Hello World</hello>
  }
`);

transform("pass-though capital (component) tags", `
  () => {
    <Hello>Hello World</Hello>
  }
`);

transform("pass-through HTML tags", `
  () => {
    <div>
      <uncommon />
      <blockquote />
      <input />
      <h1 />
      <h2 />
      <h3 />
      <h4 />
      <h5 />
      <h6 />
      <p />
      <a />
      <ul />
      <ol />
      <li />
      <i />
      <b />
      <em />
      <strong />
      <span />
      <hr />
      <img />
      <div />
      <br />
    </div>
  }
`);

transform("Handle member-type tags", `
  () => {
    <Foo.Bar />
  }
`);

transform("Member property implemented as name", `
  () => {
    Bar: {
      color: blue;
    }

    <Foo.Bar />
  }
`);