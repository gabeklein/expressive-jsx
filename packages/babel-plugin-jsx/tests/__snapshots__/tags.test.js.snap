// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handle member-type tags 1`] = `  () => <Foo.Bar />;`;

exports[`Member property implemented as name 1`] = `
  () => <Foo.Bar className="Bar_1nq" />;
  css(\`
  	.Bar_1nq {
  		color: blue;
  	}
  \`);
`;

exports[`pass-though capital (component) tags 1`] = `  () => <Hello>Hello World</Hello>;`;

exports[`pass-through HTML tags 1`] = `
  () => (
    <div>
      <div />
      <blockquote />
      <input />
      <h1 />
      <h2 />
      <h3 />
      <h4 />
      <h5 />
      <h6 />
      <p />
      <a />
      <ul />
      <ol />
      <li />
      <i />
      <b />
      <em />
      <strong />
      <span />
      <hr />
      <img />
      <div />
      <br />
    </div>
  );
`;

exports[`strip tag; emit default 1`] = `  () => <div>Hello World</div>;`;

exports[`will convert \`./../file.ext\` to require 1`] = `
  () => (
    <img src={require("./logo.svg")}>
      This should be a require-sourced file!
    </img>
  );
`;
